src/unix/linux-syscalls.c

```diff
@@ -446,7 +446,7 @@ int uv__utimesat(int dirfd,
 
 ssize_t uv__preadv(int fd, const struct iovec *iov, int iovcnt, off_t offset) {
 #if defined(__NR_preadv)
-  return syscall(__NR_preadv, fd, iov, iovcnt, offset);
+  return syscall(__NR_preadv, fd, iov, iovcnt, (long)offset, (long)(offset>>32));
```

Comment:
Two issues:

1. You would normally only do the split on architectures that need it (although it probably works by accident now, the last argument is simply ignored on architectures that don't need the split.)

2. The shift is technically UB on platforms where `off_t` is `int32_t`.  We build with `_FILE_OFFSET_BITS=64` so that shouldn't be a problem but maybe change the type of `offset` to `int64_t` just to be sure.

Also, spaces around the `>>` operator. :-)